### ISSUE TITLE
[RFC] add GET support to rpc specification

### ISSUE THREAD

**Issue by sakulstra** [blitz-js/blitz#2162]
### What do you want and why?
I want to be able to query(and perhaps even mutate) RPC via `GET` instead `POST` as it unlocks cdn side caching.
This is quite useful for public apis which rarely change, or are processing intensive(swr, rate-limit¹).
Considering cdn a rate-limiting solution(which it isn't), one might even argue that `GET` might be useful for mutations as well.

### Possible implementation(s)
**Always on/Opt in/ Opt out/ per query configuration**
I guess there are multiple ways one could approach this:
1. One could expose queries as `GET/POST` by default and let the consumer choose. `useQuery` would default to  the `POST` way and only the "advanced user", who knows what it is doing set's some property on a per `useQuery` basis to make it use `GET` instead. I'm not very familiar with the blitz source yet - on a first look it seems like we might be able to just add a handler for get requests [alongside the post handler](https://github.com/blitz-js/blitz/blob/canary/packages/core/src/server/rpc-server.ts#L22)?
2. We could enable/disable the `GET` way per default and have a config flag to enable it?
3. One could imagine another of these named export via convention next uses a lot like `export const METHOD = 'POST/GET';` on a query level?
4. ?

### Additional context
Splitting this off of https://github.com/blitz-js/blitz/discussions/1777#discussioncomment-486342 as this discussion is about  different things (the general specification) and I'd like to focus this ticket only on the `GET` support.

**Summing up the points over there**
- Initial idea was to have `POST` for mutations and `GET` for queries[*](https://github.com/blitz-js/blitz/discussions/1777#discussioncomment-303438), but sadly it's not that easy as authenticated queries should probably not be cached on cdn as they might include sensitive info[*](https://github.com/blitz-js/blitz/discussions/1777#discussioncomment-303446).
- GET has more strict size limitations than POST, which has some variance throughout browsers, but is +8k chars on evergreens.[*](https://github.com/blitz-js/blitz/discussions/1777#discussioncomment-303440)

¹I've seen apps use CDN caching headers as soft rate limit solution, but I guess it doesn't properly work like that as it's usually caching per CDN region, not per lambda.

**Comment by flybayer** [comment]
Thanks @sakulstra! I think we should make it opt-in. It's quite nice and simple when all requests use POST and the network requests all look uniform. As well because of the query string limit. Also it helps prevent user data being accidentally cached.

But I'm open to hear other's thoughts on this.

**Comment by sakulstra** [comment]
I'm in dire need of this as I don't want to put effort into redis caching or similar

I'll try to solve it via a query/mutation level named export (e.g. sth like `export const METHOD: 'POST' | 'GET' `) and then i guess the more heated discussion can continue/start in the pr :D

**Comment by flybayer** [comment]
@sakulstra I think do it like this. exporting a config constant is a pattern already used in next.js

```ts
export const config = {
  httpMethod: 'POST' | 'GET'
}
```

**Comment by embiem** [comment]
I'm also interested in this. The current PR is difficult to me to get into, because of the huge change list. Is it still being worked on?

**Comment by sakulstra** [comment]
@embiem, sorry my branch just was behind canary a lot, inflating changes (just updated).

Still just playing with blitz on the weekend so i'm progressing extremely slow.
While it's "~still being worked on" i'd be more than happy with someone chiming in/picking it up :pray:!

Iirc i got stock here: https://github.com/sakulstra/blitz/pull/1/files#diff-29a4ffa429f282d408ff1e65bf1eed83611b2a0db1b48beb6422d3b50d8743b6R229 because i didn't manage to actually access the resolver (although i think i adjusted the other parts to get it :shrug: ).

**Comment by flybayer** [comment]
It's undefined on the client because the original query file is not included in the client build: https://github.com/blitz-js/blitz/blob/canary/packages/core/src/rpc-client.ts#L200

So we'll need to take some other approach for getting that config to the client 🤔

**Comment by siddhsuresh** [comment]
@flybayer would this feature still be relevant/beneficial to the toolkit?
I have made the basic changes allow the users to opt in to the GET request in the query files. 
I wanted to confirm, because this issue is more an a year old, before continuing to work on this.

**Comment by flybayer** [comment]
@siddhsuresh yes, let's add it!

**Comment by cawabunga** [comment]
> @sakulstra I think do it like this. exporting a config constant is a pattern already used in next.js
> 
> ```ts
> export const config = {
>   httpMethod: 'POST' | 'GET'
> }
> ```

It would be cool if this had been reflected in the documentation.

```ts
import { ResolverConfig } from 'blitz'

export const config: ResolverConfig = {
  httpMethod: 'GET',
}
```
